### ISSUE TITLE
When "redirect to standard forums" is enabled, clicking the "mobile" link in forum footer automatically redirects back

### ISSUE THREAD

**Issue by penguinmoose** [ScratchAddons/ScratchAddons#5242]
### Description

If "redirect to standard forums" is enabled, clicking the "mobile" link in the forum footer will automatically redirect back.

When the "mobile" link is clicked, the change to mobile forums is intentional, so it shouldn't redirect back to the standard forums. Additionally, if a link on the mobile forums is clicked, the addon shouldn't redirect back to the normal forums. However, some people may not want this, so there might be a toggle in the addon to turn this off.

### Steps to reproduce

1. Go to the [Scratch Forums](https://scratch.mit.edu/discuss).
2. Scroll down to the footer.
3. Click the "Mobile" link, right under "Change your signature".
4. The browser will go to the mobile forums, but will immediately redirect back.

### Demonstration(s) and additional context

https://user-images.githubusercontent.com/67451414/197353008-2924edbe-6b8b-433e-a10f-e5b8b7fe261f.mp4

### Environment

Browser: Google Chrome 106.0.5249.103 (Official Build) (arm64)
Scratch Addons version: 1.28.0
Operating system: MacOS on M1 Pro

**Comment by WorldLanguages** [comment]
> When the "mobile" link is clicked, the change to mobile forums is intentional, so it shouldn't redirect back to the standard forums. Additionally, if a link on the mobile forums is clicked, the addon shouldn't redirect back to the normal forums.

I'm not sure how this could work. Both the tab where the user clicked "mobile", as well as any other links opened in new tabs from that tab, and what I just said with the new tabs themselves etc., should be marked as non-redirectable. How?

**Comment by DNin01** [comment]
We could solve that by making this addon hide the Mobile button so that the user can't even begin to switch to the Mobile view, which, seeing that they're using this addon, they probably don't need to do.

**Comment by mxmou** [comment]
> > When the "mobile" link is clicked, the change to mobile forums is intentional, so it shouldn't redirect back to the standard forums. Additionally, if a link on the mobile forums is clicked, the addon shouldn't redirect back to the normal forums.
> 
> I'm not sure how this could work. Both the tab where the user clicked "mobile", as well as any other links opened in new tabs from that tab, and what I just said with the new tabs themselves etc., should be marked as non-redirectable. How?

Could a URL parameter be added to the "Mobile" link?

**Comment by penguinmoose** [comment]
> We could solve that by making this addon hide the Mobile button so that users of this addon can't even begin to switch to the Mobile view, which, seeing that they'd be using this addon, they probably don't need to use.

I would imagine most people use this addon as a convenience, not that they never visit the mobile forums - sometimes I do just to play around with it.

**Comment by penguinmoose** [comment]
> > When the "mobile" link is clicked, the change to mobile forums is intentional, so it shouldn't redirect back to the standard forums. Additionally, if a link on the mobile forums is clicked, the addon shouldn't redirect back to the normal forums.
> 
> I'm not sure how this could work. Both the tab where the user clicked "mobile", as well as any other links opened in new tabs from that tab, and what I just said with the new tabs themselves etc., should be marked as non-redirectable. How?

One way is for Scratch Addons to store internally whether the user went to the mobile page through the mobile link, etc.

**Comment by WorldLanguages** [comment]
> Could a URL parameter be added to the "Mobile" link?

Yeah, that could work. That URL parameter should also be added to any link in the page, if the current URL also has it, to solve the issue with opening parts of the mobile forums in new tabs, without them redirecting.

> One way is for Scratch Addons to store internally whether the user went to the mobile page through the mobile link, etc.

Yes. How? It becomes complicated because you can go to the mobile view, open a forum category in a new tab, then open a forum post in a new tab again. It's not straightforward to conclude whether a tab in the "opener chain" was user-initiated - I'd say it's impossible to get the opener chain itself.

**Comment by mxmou** [comment]
This is no longer an issue because the mobile forums have been removed.
